packages/forms/src/components/DateInput/ControlledInput.tsx

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import {
+  ControllerFieldState,
+  ControllerRenderProps,
+  FieldValues,
+  UseFormStateReturn,
+} from "react-hook-form";
+import { useIntl } from "react-intl";
+
+import { dateMessages } from "@gc-digital-talent/i18n";
+
+import { DateSegment, DATE_SEGMENT } from "./types";
+import {
+  getMonthOptions,
+  getMonthSpan,
+  inputStyles,
+  setComputedValue,
+  splitSegments,
+} from "./utils";
+
+interface ControlledInputProps {
+  field: ControllerRenderProps<FieldValues, string>;
+  fieldState: ControllerFieldState;
+  formState: UseFormStateReturn<FieldValues>;
+  show: Array<DateSegment>;
+}
+
+const ControlledInput = ({
+  field: { onChange, value, name },
+  formState: { defaultValues },
+  show,
+}: ControlledInputProps) => {
+  const intl = useIntl();
+  const { year, month, day } = splitSegments(
+    defaultValues ? defaultValues[name] : undefined,
+  );
+  const ID = {
+    YEAR: `${name}Year`,
+    MONTH: `${name}Month`,
+    DAY: `${name}Day`,
+  };
+
+  const handleChange = (segmentValue: string, segment: DateSegment) => {
+    const newValue = setComputedValue({
+      initialValue: value,
+      value: segmentValue,
+      segment,
+      show,
+    });
+
+    if (newValue) {
+      onChange(newValue);
+    }
+  };
+
+  const handleYearChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    handleChange(e.target.value, DATE_SEGMENT.Year);
```

Comment:
Could you pad this one to four digits, too?
![image](https://user-images.githubusercontent.com/8978655/225641539-24236f04-f284-4264-8fcc-43e043af031a.png)

Reply:
Padded year in [pad year value to 4 digits](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5928/commits/95373f4a751381fe160bb88f8ba83d8de76d5415)